packages/fuels-programs/src/contract.rs

```diff
@@ -858,10 +870,15 @@ impl MultiContractCallHandler {
                 Err(Error::RevertTransactionError { receipts, .. })
                     if ContractCall::is_missing_output_variables(&receipts) =>
                 {
-                    self.contract_calls
-                        .iter_mut()
-                        .take(1)
-                        .for_each(|call| call.append_variable_outputs(1));
+                    // It doesn't matter where the missing output is added,
+                    // all calls will be merged to a single script tx
+                    self.contract_calls[0].append_variable_outputs(1);
```

Comment:
What makes you sure that `self.contract_calls.len() > 0` will always be true at this point?

If there is a reason can you add it to a comment? 

Otherwise you can do: 
```rust
                    if let Some(first_contract_call) = self.contract_calls.first_mut() {
                        first_contract_call.append_message_outputs(1);
                    }
```

Reply:
Using simulate or call requires that the underlying tx is built via `build_tx()` which fails if there are no contract calls.

Reply:
This trick should be documented

Reply:
I've applied the changes to both single and multi calls. I did change the naming a little and opted of `iter_mut().take(1)` instead of the above.

Reply:
Papi's got tricks up his sleeves.